src/r_brightmaps.c

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright(C) 2024 Roman Fomin
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "doomtype.h"
+#include "info.h"
+#include "m_misc.h"
+#include "m_swap.h"
+#include "r_defs.h"
+#include "v_fmt.h"
+#include "w_wad.h"
+#include "z_zone.h"
+
+#define MAX_FRAMES 29
+#define MAX_ROTATIONS 9
+
+boolean brightmaps;
+
+static patch_t *spr_brightmaps[NUMSPRITES][MAX_FRAMES][MAX_ROTATIONS];
```

Comment:
That's 261 pointers per sprite, even if no brightmaps are ever loaded. Should we turn this into dynamic allocation?

Reply:
Sure, this is temporary, for simplification. By the way, I'm not sure if the frames/rotations are working correctly.